modules/reactor/src/meshgenerators/ConcentricCircleGeneratorBase.C

```diff
@@ -0,0 +1,275 @@
+    if (_ring_radii[i] <= _ring_radii[i - 1])
+      paramError("ring_radii", "This parameter must be strictly ascending.");
+  if (_ring_radii.size() != _ring_inner_boundary_layer_params.widths.size() ||
+      _ring_radii.size() != _ring_inner_boundary_layer_params.intervals.size() ||
+      _ring_radii.size() != _ring_inner_boundary_layer_params.biases.size() ||
+      _ring_radii.size() != _ring_outer_boundary_layer_params.widths.size() ||
+      _ring_radii.size() != _ring_outer_boundary_layer_params.intervals.size() ||
+      _ring_radii.size() != _ring_outer_boundary_layer_params.biases.size())
+    paramError("ring_radii",
+               "The inner and outer ring boundary layer parameters must have the same sizes as "
+               "ring_radii.");
+}
+
+void
+ConcentricCircleGeneratorBase::assignInterfaceBoundaryNames(ReplicatedMesh & mesh)
+{
+  if (!_inward_interface_boundary_names.empty())
+  {
+    for (unsigned int i = 0; i < _inward_interface_boundary_names.size(); i++)
+    {
+      mesh.get_boundary_info().sideset_name(i * 2 + 2 + _interface_boundary_id_shift) =
+          _inward_interface_boundary_names[i];
+      mesh.get_boundary_info().nodeset_name(i * 2 + 2 + _interface_boundary_id_shift) =
+          _inward_interface_boundary_names[i];
+    }
+  }
+  if (!_outward_interface_boundary_names.empty())
+  {
+    for (unsigned int i = 0; i < _outward_interface_boundary_names.size(); i++)
+    {
+      mesh.get_boundary_info().sideset_name(i * 2 + 1 + _interface_boundary_id_shift) =
+          _outward_interface_boundary_names[i];
+      mesh.get_boundary_info().nodeset_name(i * 2 + 1 + _interface_boundary_id_shift) =
+          _outward_interface_boundary_names[i];
+    }
+  }
+}
+
+void
+ConcentricCircleGeneratorBase::assignBlockIdsNames(ReplicatedMesh & mesh,
+                                                   std::vector<subdomain_id_type> & block_ids_old,
+                                                   std::vector<subdomain_id_type> & block_ids_new,
+                                                   std::vector<SubdomainName> & block_names,
+                                                   const std::string & generator_name)
+{
+  for (auto it = block_names.begin(); it != block_names.end() - 1; it++)
+  {
+    auto it_tmp = std::find(block_names.begin(), it + 1, *(it + 1));
+    if (it_tmp != it + 1 && block_ids_new[std::distance(block_names.begin(), it + 1)] !=
+                                block_ids_new[std::distance(block_names.begin(), it_tmp)])
+      mooseError("In ",
+                 generator_name,
+                 " ",
+                 _name,
+                 ": blocks with different ids cannot have the same block name.");
+  }
+  for (const auto & elem : mesh.element_ptr_range())
+    for (unsigned i = 0; i < block_ids_old.size(); ++i)
+      if (elem->subdomain_id() == block_ids_old[i])
+      {
+        elem->subdomain_id() = block_ids_new[i];
+        break;
+      }
```

Comment:
I wouldnt be surprised if we had this in libmesh somewhere else as a rename_block but not sure

Reply:
I looked up the doxygen for a while and did not find one.